apps/settings/js/modules/bluetooth/bluetooth_connection_manager.js

```diff
@@ -0,0 +1,674 @@
+/**
+ * BluetoothConnectionManager:
+ *   - BluetoothConnectionManager is an Observable that manage the connection
+ *     state.
+ *   - It has some observable properties: connectingAddress, connectedAddress.
+ * BluetoothConnectionManager only update state and does not involve in any UI
+ *   logic.
+ *
+ * @module BluetoothConnectionManager
+ */
+define(function(require) {
+  'use strict';
+
+  var AdapterManager = require('modules/bluetooth/bluetooth_adapter_manager');
+  var Observable = require('modules/mvvm/observable');
+  require('shared/async_storage');
```

Comment:
Suggest to use `var AsyncStorage = require('shared/async_storage');` so we can avoid using `window.AsyncStorage`.